lib/cloud_controller/dependency_locator.rb

```diff
@@ -432,11 +432,19 @@ def build_task_client
     end
 
     def build_opi_task_client
-      ::OPI::TaskClient.new(config)
+      ::OPI::TaskClient.new(config, build_task_completion_callback_generator, VCAP::CloudController::Diego::TaskEnvironmentVariableCollector)
     end
 
     def build_bbs_task_client
-      VCAP::CloudController::Diego::BbsTaskClient.new(build_bbs_client)
+      VCAP::CloudController::Diego::BbsTaskClient.new(config, build_bbs_client, build_task_recipe_builder)
+    end
+
+    def build_task_recipe_builder
```

Comment:
Would prefer to see `build_task_recipe_builder` simply be instantiated within `BbsTaskClient`. In this way, it mirrors the same setup as `BbsAppsClient` instantiating the `AppRecipeBuilder`.